exonum/src/runtime/dispatcher/error.rs

```diff
@@ -48,6 +48,10 @@ pub enum Error {
     UnauthorizedCaller = 11,
     /// Malformed arguments for calling a service interface method.
     MalformedArguments = 12,
+    /// Action requested at the inappropriate time.
+    /// E.g. action that requires mutable access to the database, requested
+    /// in the `after_commit`.
+    InappropriateTimeForAction = 13,
```

Comment:
We either allow this or make interfaces bigger (currently we have 3 methods in service *Context structures of the Rust runtime, with type-checking it will become 6 (one immutable and one mutable method)).

As an alternative, we can `panic` if action is requested at the inappropriate time. Would it be ok?